be-python-flask/Jenkinsfile.template

```diff
@@ -30,49 +30,61 @@ odsPipeline(
     '*': 'dev'
   ]
 ) { context ->
+  stageTestSuite(context, context.nexusHostWithBasicAuth, context.nexusHost.tokenize('//')[1])
+  stageScanForSonarqube(context)
   stageBuild(context)
   stageStartOpenshiftBuild(context, [nexusHostWithBasicAuth: context.nexusHostWithBasicAuth, nexusHostWithoutScheme: context.nexusHost.tokenize('//')[1]])
   stageDeployToOpenshift(context)
 }
 
+def stageTestSuite(def context, def nexusHostWithBasicAuth, def nexusHostWithoutScheme) {
+  String testLocation = 'build/test-results/test'
+  String coverageLocation = 'build/test-results/coverage'
+
+  stage('Prepare Test Suite') {
+    sh """
+      virtualenv testsuite
+      . ./testsuite/bin/activate
+      pip install -i ${nexusHostWithBasicAuth}/repository/pypi-all/simple --trusted-host ${nexusHostWithoutScheme} -r tests/requirements.txt
+      mkdir -p ${testLocation}
+      mkdir -p ${coverageLocation}
+    """
+  }
+
+  stage('Lint') {
```

Comment:
As the application grows installing the dependencies for testing will take more time. Linting does not typically require that these dependencies are installed. An idea is to move linting in earlier stage so builds fail faster and perhaps even adding the linter in the jenkins slave image.